File: frontend/src/components/Verify.js
import React, { useState } from 'react';

function Verify() {

    var loginName, verifCode;

    const [message, setMessage] = useState('');

    const doVerification = async event => {     
        
        const app_name = 'hivemindg26';
        function buildPath(route)
        {
            if (process.env.NODE_ENV === 'production') 
            {
                return 'https://' + app_name +  '.herokuapp.com/' + route;
            }
            else
            {        
                return 'http://localhost:5000/' + route;
            }
        }
     
        event.preventDefault();        
        var obj = {username:loginName.value,validationcode:verifCode.value};        
        var js = JSON.stringify(obj);        
        
        try {                
            const response = await fetch(buildPath('api/validateCode'), {method:'POST',body:js,headers:{'Content-Type': 'application/json'}});        
            var res = JSON.parse(await response.text());
            if ( res.Error === "User not found" ) {                
                setMessage(res.Error);            
            } else if (res.Error === "Incorrect validation code") {            
                setMessage(res.Error);               
            } else if (res.Error === "Validation success"){               
                setMessage('');               
                window.location.href = '/';            
            }        
        } catch(e) {            
            alert(e.toString());            
            return;        
        }        
    }; 

    return (    
        <div>
            <form className="login-form" onSubmit={ doVerification }>
                <span style={{color: "black"}}>Please check your email for your Verification Code.</span>
                <hr/>
                <input className="verif-input" type="text" id="loginName" placeholder="Username"   ref={(c) => loginName = c} />
                <input className="verif-input" type="test" id="verifCode" placeholder="Verification"   ref={(c) => verifCode = c} />
                <div className="verifButton">
                    <input type="submit" className="buttons" value="Verify" onClick={doVerification} />       
                </div>
            </form>
        </div>
    );
};

export default Verify;

File: startup_namer/lib/verify.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class verify{
  String error;

  verify({this.error});

  // factory keyword is used when the constructor doesn't always create an object
  factory verify.fromJson(Map<String, dynamic> json) {
    return verify(
        error: json['Error']
    );
  }
}

Future<verify> fetchverify(String username, String validationCode) async{
  final http.Response response = await http.post(
    'https://hivemindg26.herokuapp.com/api/validateCode',
    headers: <String, String>{
      'Content-Type': 'application/json; charset=UTF-8'
    },
    body: jsonEncode(<dynamic, dynamic>{
      'username': username,
      'validationcode': validationCode
    }),
  );

  if (response.statusCode == 200) {
    // print(response.body);
    return verify.fromJson(jsonDecode(response.body));
  }
  else{
    throw Exception('Failed to load');
  }
}